dips/dip-182.md

```diff
@@ -0,0 +1,71 @@
+---
+dip: 182
+title: On-Chain Domain Resolution
+authors: Sunmi Lee (@sunmilee), David Wolinsky (@davidiw)
+status: Draft
+type: Informational
+created: 06/08/2021
+last_updated: 06/015/2021
+issue: https://github.com/diem/dip/issues/182
+---
+
+# Summary
+
+This DIP proposes adding VASP domain names on-chain and describes the domain name resolution process.
+
+On-chain domain name resolution allows organizations to define their product names, and for VASPs to look up other VASPs using the product names.
+Having a human-readable VASP identifier means we can enumerate all VASPs by their names, which is useful for merchant purposes as well as p2p transactions.
+For example, given a VASP domain name in a payment transaction, VASPs can look up where to route the payment correctly using an approved mapping of VASP identifiers to on-chain addresses.
+
+# On-chain data
+
+## Domain Format
+A VASP domain name is a unique string that is mapped to a VASP. Specification:
+* Case insensitive
+* Valid regular expression: `^[a-zA-Z0-9][a-zA-Z0-9.]*$`
+* Maximum length: 63 characters
+
+## VASP Domain
+
+```
+resource struct VaspDomains {
+    domains: vector<VaspDomain>,
+}
+struct VaspDomain {
+    domain: vector<u8>,  // Utf-8 encoded
+}
+```
+* Field definitions:
+    * `domain` - name of a VASP domain
+* The `VaspDomains` resource can only be published into an account with `Role == PARENT_VASP_ROLE_ID`
+* The `VaspDomains` contains a list of `VaspDomain` structs that are associated with a VASP. It is possible to register more than one VaspDomain for a given VASP
+* Only the Treasury Compliance (TC) account (address `0xB1E55ED`) can manipulate a `VaspDomains` resource:
+    * Only TC account can create and publish a `VaspDomains` resource
+    * Only TC account can add, remove or update a `VaspDomain` within `VaspDomains` resource
+* `VaspDomains` resource will be created in an empty state as part of creating a `ParentVASP` account resource, and existing `ParentVASP` accounts without `VaspDomains` will have the resource instantiated by the DiemRoot account
+* In order to register a VASP domain, a VASP needs to submit a request to Diem Association. After approval, Diem Association will use TC account to issue an on-chain transaction to register a VaspDomain. Diem Association will ensure uniqueness of domain names across all VASPs.
```

Comment:
If the number of VASPs is not too large, it should be fine for the Diem Association to handle registration, expiration, uniqueness, re-use, and other characteristics.  Before this DIP can move to Accepted, I think we will need to have commitment from the Diem Association that these responsibilities can be taken on.

Reply:
Who do we need to talk to ? @aching @davidiw

Reply:
I think this should be pretty quick conversation with @pdhamdhere .